src/bun.js/bindings/bindings.cpp

```diff
@@ -893,25 +936,43 @@ JSC__JSValue WebCore__FetchHeaders__createValue(JSC__JSGlobalObject* arg0, Strin
     }
 
     Ref<WebCore::FetchHeaders> headers = WebCore::FetchHeaders::create();
-    headers->fill(WebCore::FetchHeaders::Init(WTFMove(pairs)));
+    WebCore::propagateException(*arg0, throwScope, 
+        headers->fill(WebCore::FetchHeaders::Init(WTFMove(pairs)))
+    );
     pairs.releaseBuffer();
     return JSC::JSValue::encode(WebCore::toJSNewlyCreated(arg0, reinterpret_cast<Zig::GlobalObject*>(arg0), WTFMove(headers)));
 }
-void WebCore__FetchHeaders__get_(WebCore__FetchHeaders* headers, const ZigString* arg1, ZigString* arg2)
+void WebCore__FetchHeaders__get_(WebCore__FetchHeaders* headers, const ZigString* arg1, ZigString* arg2, JSC__JSGlobalObject* global)
 {
-    *arg2 = Zig::toZigString(headers->get(Zig::toString(*arg1)).releaseReturnValue());
+    auto throwScope = DECLARE_THROW_SCOPE(global->vm());
+    auto result = headers->get(Zig::toString(*arg1));
+    if (result.hasException())
+        WebCore::propagateException(*global, throwScope, result.releaseException());
+    else
+        *arg2 = Zig::toZigString(result.releaseReturnValue());
 }
-bool WebCore__FetchHeaders__has(WebCore__FetchHeaders* headers, const ZigString* arg1)
+bool WebCore__FetchHeaders__has(WebCore__FetchHeaders* headers, const ZigString* arg1, JSC__JSGlobalObject* global)
 {
-    return headers->has(Zig::toString(*arg1)).releaseReturnValue();
+    auto throwScope = DECLARE_THROW_SCOPE(global->vm());
+    auto result = headers->has(Zig::toString(*arg1));
+    if (result.hasException())
+        WebCore::propagateException(*global, throwScope, result.releaseException());
```

Comment:
we need to return a value even if there is an exception

Reply:
Good catch, I'm too used to Zig catching this kind of thing.